services/src/test/java/io/grpc/protobuf/services/HealthCheckingLoadBalancerFactoryTest.java

```diff
@@ -402,37 +438,48 @@ public void healthCheckDisabledWhenServiceNotImplemented() {
     // In reality UNIMPLEMENTED is generated by GRPC server library, but the client can't tell
     // whether it's the server library or the service implementation that returned this status.
     serverCall0.responseObserver.onError(Status.UNIMPLEMENTED.asException());
-    inOrder.verify(mockStateListeners[0]).onSubchannelState(
-        eq(ConnectivityStateInfo.forNonError(READY)));
+    inOrder.verify(hasHealthConsumer ? mockHealthListeners[0] : mockStateListeners[0])
+        .onSubchannelState(eq(ConnectivityStateInfo.forNonError(READY)));
     assertThat(subchannels[0].logs).containsExactly(
         "ERROR: Health-check disabled: " + Status.UNIMPLEMENTED,
         "INFO: READY (no health-check)").inOrder();
 
     // subchannels[1] has normal health checking
     serverCall1.responseObserver.onNext(makeResponse(ServingStatus.NOT_SERVING));
-    inOrder.verify(mockStateListeners[1]).onSubchannelState(
+    inOrder.verify(hasHealthConsumer ? mockHealthListeners[1] : mockStateListeners[1])
+        .onSubchannelState(
         unavailableStateWithMsg("Health-check service responded NOT_SERVING for 'BarService'"));
 
     // Without health checking, states from underlying Subchannel are delivered directly to the mock
     // listeners.
     deliverSubchannelState(0, ConnectivityStateInfo.forNonError(IDLE));
     inOrder.verify(mockStateListeners[0]).onSubchannelState(
         eq(ConnectivityStateInfo.forNonError(IDLE)));
+    if (hasHealthConsumer) {
+      inOrder.verify(mockHealthListeners[0]).onSubchannelState(
+          eq(ConnectivityStateInfo.forNonError(IDLE)));
+    }
 
     // Re-connecting on a Subchannel will reset the "disabled" flag.
     assertThat(healthImpls[0].calls).hasSize(0);
     deliverSubchannelState(0, ConnectivityStateInfo.forNonError(READY));
+    if (hasHealthConsumer) {
```

Comment:
Don't really need the if condition for this check.

Reply:
Probably should add checks that the old path is still behaving the old way.

Reply:
oh i why you asked this. Actually I used Parameterized test that covers both cases (see L108).